src/python_testing/TC_CADMIN_1_5.py

```diff
@@ -122,11 +169,16 @@ async def test_TC_CADMIN_1_5(self):
 
         self.step(2)
         params = await self.open_commissioning_window(dev_ctrl=self.th1, timeout=180, node_id=self.dut_node_id)
+        logging.info(f"Commissioning window params: {params}")
 
         self.step(3)
-        services = await self.get_txt_record()
-        if services.txt_record['CM'] != "2":
-            asserts.fail(f"Expected cm record value not found, instead value found was {str(services.txt_record['CM'])}")
+        # Wait for DNS-SD advertisement with correct CM value and discriminator
+        # This will either return a valid service or assert failure
+        service = await self.wait_for_correct_cm_value(
+            expected_cm_value="2",
```

Comment:
```suggestion
            expected_cm_value=2,
```

Seems better since spec seems to have these integers

Reply:
This does seem better, the DNS-SD returns the values as strings, but have changed to handle this inside the function.